azure_ai_search_rag.ipynb

```diff
@@ -0,0 +1,908 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Estimate Cost for Embedding Generation\n",
+    "\n",
+    "As per the information from [Lunary.ai's Mistral Tokenizer](https://lunary.ai/mistral-tokenizer), one token is approximately equivalent to five characters of text. \n",
+    "\n",
+    "According to [Mistral's Pricing](https://mistral.ai/technology/#pricing), the cost for using `mistral-embed` is $0.1 per 1M tokens for both inputs and outputs.\n",
+    "\n",
+    "In the following code block, we will calculate the estimated cost for generating embeddings based on the size of our dataset and these pricing details."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 54,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Estimated cost for generating embeddings: $0.19047898000000002\n"
+     ]
+    }
+   ],
+   "source": [
+    "def estimate_cost(data, cost_per_million_tokens=0.1):\n",
+    "    total_characters = sum(len(entry['content']) for entry in data)\n",
+    "    total_tokens = total_characters / 5  # 1 token is approximately 5 characters\n",
+    "    total_tokens_in_millions = total_tokens / 1_000_000\n",
+    "    total_cost = total_tokens_in_millions * cost_per_million_tokens\n",
+    "    return total_cost\n",
+    "\n",
+    "estimated_cost = estimate_cost(data)\n",
+    "print(f\"Estimated cost for generating embeddings: ${estimated_cost}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Transform Dataset for Azure AI Search Upload"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Function to transform your dataset into the format required by Azure AI Search\n",
+    "def transform_to_search_document(record):\n",
+    "    return {\n",
+    "        \"id\": record[\"id\"],\n",
+    "        \"arxiv_id\": record[\"arxiv_id\"],\n",
+    "        \"title\": record[\"title\"],\n",
+    "        \"content\": record[\"content\"]\n",
+    "    }\n",
+    "    \n",
+    "# Transform all documents in the dataset\n",
+    "transformed_documents = [transform_to_search_document(doc) for doc in data]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Generate Embeddings"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 58,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def generate_embeddings(documents, model):\n",
+    "    for document in documents:\n",
+    "        content = document['content']\n",
+    "        embed = mistral.embeddings(\n",
+    "            model=model, input=[content]\n",
+    "        )\n",
+    "        # Add the embedding to the document\n",
+    "        document['embedding'] = embed.data[0].embedding\n",
+    "    return documents\n",
```

Comment:
You;'re changing the objects in the documents list in place, not sure why return it and re-assign it